pkg/envoy/lds/rbac.go

```diff
@@ -75,41 +68,20 @@ func (lb *listenerBuilder) buildInboundRBACPolicies() (*xds_network_rbac.RBAC, e
 }
 
 // buildRBACPolicyFromTrafficTarget creates an XDS RBAC policy from the given traffic target policy
-func buildRBACPolicyFromTrafficTarget(trafficTarget trafficpolicy.TrafficTargetWithRoutes) (*xds_rbac.Policy, error) {
-	policy := &rbac.Policy{}
+func buildRBACPolicyFromTrafficTarget(trafficTarget trafficpolicy.TrafficTargetWithRoutes) *xds_rbac.Policy {
+	pb := &rbac.PolicyBuilder{}
 
 	// Create the list of principals for this policy
-	var principalRuleList []rbac.RulesList
 	for _, downstreamPrincipal := range trafficTarget.Sources {
-		principalRule := rbac.RulesList{
-			OrRules: []rbac.Rule{
-				{Attribute: rbac.DownstreamAuthPrincipal, Value: downstreamPrincipal.String()},
-			},
-		}
-		principalRuleList = append(principalRuleList, principalRule)
+		pb.AddPrincipal(downstreamPrincipal.String())
 	}
-	policy.Principals = principalRuleList
-
 	// Create the list of permissions for this policy
-	var permissionRuleList []rbac.RulesList
 	for _, tcpRouteMatch := range trafficTarget.TCPRouteMatches {
 		// Matching ports have an OR relationship
-		var orPortRules []rbac.Rule
 		for _, port := range tcpRouteMatch.Ports {
-			portRule := rbac.Rule{
-				Attribute: rbac.DestinationPort, Value: strconv.Itoa(port),
-			}
-			orPortRules = append(orPortRules, portRule)
+			pb.AddAllowedDestinationPort(port)
```

Comment:
How do we know from the API whether these rules are using AND or OR? Previously this was clear by the variable name `orPortRules`

Reply:
in case this is referring to teh API on the builder, it is the responsbility of the caller. So AddAllowedDestinationPort, would be called, and how they are configured depends on if the caller also calls `UseAndForPermissions`. 

It may not be perfect, but my hope is its clear enough, while being simple enough we can make future changes if confusion arises

Reply:
resolving, feel free to re-open if more discussion is necessary!